### PR TITLE
OpenAPI different default content type for pojo return and primitives

### LINKED ISSUE #34700: OpenAPI generation content-type mismatch
### Describe the bug

Let's consider the following endpoint (using spring compatibility, but it should not matter):

```java
    @GetMapping("/hello")
    public String hello() {
        return "hello world";
    }
```

The generated OpenAPI document has the following content:

```yaml
---
openapi: 3.0.3
info:
  title: fortune-demo API
  version: 1.0.0-SNAPSHOT
paths:
  /hello:
    get:
      tags:
      - Hello Controller
      responses:
        "200":
          description: OK
          content:
            application/json:
              schema:
                type: string
components:
  securitySchemes:
    SecurityScheme:
      type: http
      description: Authentication
      scheme: basic

```

Two issues:

1. I didn't setup anything related to security - so why it's there?
2. the content type is wrong, it should be plain text in this case

(2) leads to be unable to use the Swagger UI as it gets a 405 HTTP response. 

### Expected behavior

It should provide the right content type and no security description.

### Actual behavior

See description.

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by phillip-kruger**
Fix #34700

Draft for now until OpenAPI release

### CODE REVIEW

**Review by geoand — APPROVED** [review]

File: extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java [hunk 1]
```diff
@@ -164,6 +164,8 @@ public class SmallRyeOpenApiProcessor {
     static {
         System.setProperty(io.smallrye.openapi.api.constants.OpenApiConstants.DEFAULT_PRODUCES, "application/json");
         System.setProperty(io.smallrye.openapi.api.constants.OpenApiConstants.DEFAULT_CONSUMES, "application/json");
+        System.setProperty(io.smallrye.openapi.api.constants.OpenApiConstants.DEFAULT_PRODUCES_PRIMITIVES, "plain/text");
+        System.setProperty(io.smallrye.openapi.api.constants.OpenApiConstants.DEFAULT_CONSUMES_PRIMITIVES, "plain/text");
```
Comment (rsvoboda) [inline extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java:168]: This should be `"text/plain"`

**Comment by rsvoboda** [comment]
@phillip-kruger also shouldn't this have noteworthy label as this changes the default?

File: extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java (hunk 1 above)
Comment (rsvoboda) [reply extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java:168]: Created https://github.com/quarkusio/quarkus/pull/36247

File: extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/HttpSecurityProcessor.java [hunk 2]
```diff
@@ -216,7 +215,9 @@ SyntheticBeanBuildItem initBasicAuth(
                 && !buildTimeConfig.auth.basic.orElse(false)) {
             //if not explicitly enabled we make this a default bean, so it is the fallback if nothing else is defined
             configurator.defaultBean();
-            securityInformationProducer.produce(SecurityInformationBuildItem.BASIC());
+            if (buildTimeConfig.auth.basic.isPresent() && buildTimeConfig.auth.basic.get()) {
```
Comment (michalvavrik) [inline extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/HttpSecurityProcessor.java:218]: line 215 `!buildTimeConfig.auth.basic.orElse(false)` means that

`quarkus.http.auth.basic` is missing => inside IF body
`quarkus.http.auth.basic=false` => inside IF body
`quarkus.http.auth.basic=true` => don't go in

while line 218 says

`buildTimeConfig.auth.basic.isPresent()` => `quarkus.http.auth.basic=false` || `quarkus.http.auth.basic=true`
`buildTimeConfig.auth.basic.get()` => `quarkus.http.auth.basic=true`

so in order to have produced `BASIC` here it means that `quarkus.http.auth.basic` must be both `true` and `false`

@phillip-kruger I don't know what was intention of this condition, can you please fix it? Thank you

**Pull request merged**